cpp/src/parquet/types.cc

```diff
@@ -464,7 +464,10 @@ std::shared_ptr<const LogicalType> LogicalType::FromThrift(
   } else if (type.__isset.FLOAT16) {
     return Float16LogicalType::Make();
   } else {
-    throw ParquetException("Metadata contains Thrift LogicalType that is not recognized");
+    // Or provide some mechanism to optionally error here?
```

Comment:
I agree that we might need an option to make it throw or ignore for unknown logical type.

BTW, it seems that parquet-java also fails on unknown type: https://github.com/apache/parquet-java/blob/6809a18597e56709696af58313343aeea93f0b99/parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java#L1142-L1179

Reply:
Should we regard it as physical type with unknown logical type

Reply:
UNKNOWN logical type means all values are null, IIUC.

Reply:
I also prefer `UndefinedLogicalType`. I checked arrow-rs that it also raise error in this condition

Reply:
https://github.com/apache/arrow/blob/13630c7a8316bccce4a119d193e3a46f5a9a35bc/cpp/src/parquet/metadata.cc#L1920

You might need to hack the thrift object directly from `metadata_->schema` above by stetting an unknown logical value. This means we have to create a parquet file any way.

Reply:
> I still have to figure out how to write a full-on file so that this is tested properly! (I figured out how to test the schema conversion but not how to check a full-on file yet).

Probably hacking the generated thrift code with a very high field ID and writing out the file and then trying to read it back in could work?